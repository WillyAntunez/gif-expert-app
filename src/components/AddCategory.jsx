import { useState } from 'react';

export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setInputValue] = useState('');

    const onInputChange = (e) => {
        setInputValue(e.target.value);
    };

    const onSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 0) {
            // setCategories((categories) => [...categories, inputValue]);
            onNewCategory(inputValue);
        }
        setInputValue('');
    };

    return (
        <form onSubmit={onSubmit}>
            <input
                type="text"
                placeholder="buscar gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    );
};
