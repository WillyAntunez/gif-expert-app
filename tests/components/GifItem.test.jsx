import React from 'react';
import { GifItem } from '../../src/components/GifItem';
import { render, screen } from '@testing-library/react';

describe('Pruebas en el componente GifItem', () => {
    const exampleId = '12345';
    const exampleUrl = 'http://ejemplo.com/api/v2/1234';
    const exampleTitle = 'title';

    test('debe de hacer match con el snapshot', () => {
        const { container } = render(
            <GifItem id={exampleId} url={exampleUrl} />
        );

        expect({ container }).toMatchSnapshot();
    });

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render(
            <GifItem id={exampleId} url={exampleUrl} title={exampleTitle} />
        );

        /* const src = screen.getByRole('img').src;
        const alt = screen.getByRole('img').alt; */

        const { src, alt } = screen.getByRole('img');

        expect(src).toBe(exampleUrl);
        expect(alt).toBe(exampleTitle);
    });

    test('debe de mostrar el titulo en el componente', () => {
        render(
            <GifItem id={exampleId} url={exampleUrl} title={exampleTitle} />
        );
        expect(screen.getByText(exampleTitle)).toBeTruthy();
    });
});
