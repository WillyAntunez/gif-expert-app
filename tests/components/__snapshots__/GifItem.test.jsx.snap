// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifItem debe de hacer match con el snapshot 1`] = `
{
  "container": <div>
    <div
      class="card"
    >
      <img
        src="http://ejemplo.com/api/v2/1234"
      />
      <p />
    </div>
  </div>,
}
`;
